Deployment Guides/Provisioning-Islandora-on-Ubuntu-16.md

```diff
@@ -0,0 +1,1788 @@
+
+Configure Tomcat (based on http://dev.digibess.it/doku.php?id=reloaded:be_base).
+
+We create directories for the repository data outside of Fedora home, so that it is easier in future to expand storage, or to use a SAN. This follows the architecture of DigiBESS.
+```
+mkdir /usr/local/fedora
+
+mkdir -p /srv/fedora/data
+
+mkdir /srv/fedora/tmp
+
+chown -R $FEDORA_USER:$FEDORA_USER /usr/local/fedora/ /srv/fedora/
+```
+
+Update Tomcat environment variables and JAVA OPTS:
+```
+sed -i s/^JAVA_OPTS/#JAVA_OPTS/ /etc/default/tomcat7
+
+cat >> /etc/default/tomcat7 <<END_JT
+JAVA_HOME=JAVA_HOME_T
+JAVA_OPTS="JAVA_OPTS_T"
+FEDORA_HOME=FEDORA_HOME_T
+END_JT
+
+sed -i "s|JAVA_HOME_T|$JAVA_HOME|g" /etc/default/tomcat7
+sed -i "s|JAVA_OPTS_T|$JAVA_OPTS|g" /etc/default/tomcat7
+sed -i "s|FEDORA_HOME_T|$FEDORA_HOME|g" /etc/default/tomcat7
+
+cat >> /etc/default/tomcat7 <<END_JT2
+PATH=$FEDORA_HOME/server/bin:$FEDORA_HOME/client/bin:$JAVA_HOME/bin:$PATH
+END_JT2
+```
+ 
+**Note:** make sure you check `/etc/default/tomcat7` to ensure everything has been generated properly. If it didn’t you may have closed your shell you may need to source `islandora-install.properties` again. 
+
+For example: 
+```
+JAVA_HOME=/usr/lib/jvm/java-8-oracle/jre
+JAVA_OPTS="-Xms3993m -Xmx3993m -Djavax.net.ssl.trustStore=/usr/local/fedora/server/truststore -Djavax.net.ssl.trustStorePassword=tomcat"
+FEDORA_HOME=/usr/local/fedora
+```
+
+Make sure the `tomcat7` service is enabled and active:
+```
+systemctl status tomcat7
+
+systemctl start tomcat7
+```
+
+If this fails, it could well be due to the memory settings, or other factors such as the specification of directories that don't yet exist. Replace the first `JAVA_OPTS` line in `/etc/default/tomcat7` with something simple, like `JAVA_OPTS="-Djava.awt.headless=true -Xmx128m"`, to verify that Tomcat runs. Then enable the real value again and stop the service: `systemctl stop tomcat7`.
+
+Optionally, add a Tomcat user for the admin interface. This should be disabled once testing is complete.
+
+Edit `/etc/tomcat7/tomcat-users.xml`:
+```
+
+<tomcat-users>
+  [ ... ]
+  <user username="tomcat" password="somepassword" roles="admin-gui,manager-gui"/>
+</tomcat-users>
+```
+
+#### Blazegraph  <a id="blazegraph"></a>
+
+This step is optional. Instructions are mostly a copy of http://dev.digibess.it/doku.php?id=reloaded:be_blazeg, with a few modifications. We have to install under a separate Tomcat from Fedora, as Fedora cannot be running when rebuilding the Resource Index (see https://github.com/discoverygarden/trippi-sail/wiki/Repacing-Mulgara-with-Blazegraph). This also makes migrating this service to a separate machine easier.
+
+Install and configure a second Tomcat.
+```
+cd /usr/share
+
+wget http://apache.mirror.vexxhost.com/tomcat/tomcat-7/v7.0.86/bin/apache-tomcat-7.0.86.tar.gz
+
+tar xzf apache-tomcat-7.0.86.tar.gz && rm -f apache-tomcat-7.0.86.tar.gz
```

Comment:
Should probably use variables for version numbers. Makes it easier to update Tomcat version etc.

Reply:
I added a note about getting the most recent. It's not straightforward to keep up with the
Apache site URLs. Let me know fi you think this is sufficient, or have a preferred approach.